src/components/disclosure/index.tsx

```diff
@@ -108,13 +116,27 @@ const Disclosure = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isOpen])
 
-  // build the className prop passed the `children` container
-  let containerClasses
-  if (typeof containerClassName === 'function') {
-    containerClasses = classNames(s.root, containerClassName(isOpen))
-  } else {
-    containerClasses = s.root
-  }
+  // if enabled, close the disclosure on click outside
+  useOnClickOutside(
+    [disclosureRef],
+    closeDisclosure,
+    closeOnClickOutside && isOpen
+  )
+
+  // if enabled, close the disclosure on focus outside
+  useOnFocusOutside(
+    [disclosureRef],
+    closeDisclosure,
+    closeOnFocusOutside && isOpen
+  )
+
+  // build the className prop to pass the `children` container
+  const containerClasses = classNames(
+    s.root,
+    typeof containerClassName === 'function'
+      ? containerClassName(isOpen)
+      : containerClassName
+  )
```

Comment:
🗒️ This also had a bug. When I updated `containerClassName` to allow either a function or a string, I did not actually make the `else` branch append `containerClassName`. Only `s.root` was being applied!